doc/classes/Array.xml

```diff
@@ -234,7 +234,9 @@
 			<argument index="0" name="value" type="Variant">
 			</argument>
 			<description>
-				Removes the first occurrence of a value from the array.
+				Removes the first occurrence of a value from the array. To remove an element by index, use [method remove] instead.
+				[b]Note:[/b] This method acts in-place and doesn't return a value. Make sure not to assign its result to a variable when using it. Instead, call [method erase] on its own statement.
```

Comment:
I'm not sure the part "Make sure not to assign its result to a variable when using it. Instead, call [method erase] on its own statement." is useful, wouldn't GDScript at least raise a warning when assign a variable with a method that returns `void`? And C# would likely not compile.

Reply:
That makes sense. Should I leave the first part of the note?

Reply:
Yeah I think it might be useful to make users aware of the return type, at least for these specific methods which in other APIs might return the removed index or value.

I also checked and GDScript doesn't raise a warning for assigning the result of a void function to a variable, maybe we'd want to add that. But I still think the warning is not necessary as it doesn't actually cause issues to do `var res = arr.erase(2)`, you just get a `res == null` but the `arr` modification still happens fine.